File: test-serenity2.Web/Modules/Games/Games/GamesColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace test_serenity2.Games.Columns;

[ColumnsScript("Games.Games")]
[BasedOnRow(typeof(GamesRow), CheckNames = true)]
public class GamesColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Name { get; set; }
    public int Year { get; set; }
    public string PlatformName { get; set; }
    public string Genre { get; set; }
}

File: test-serenity2.Web/Modules/ServerTypes/Games/GamesColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { GamesRow } from "./GamesRow";

export interface GamesColumns {
    Id: Column<GamesRow>;
    Name: Column<GamesRow>;
    Year: Column<GamesRow>;
    PlatformName: Column<GamesRow>;
    Genre: Column<GamesRow>;
}

export class GamesColumns extends ColumnsBase<GamesRow> {
    static readonly columnsKey = 'Games.Games';
    static readonly Fields = fieldsProxy<GamesColumns>();
}